### PR TITLE
digital: unified digital API

### PR DESCRIPTION
**Pull request by paulsemel**
this new version of the digital permits to add other achitecture
without changing the API.
All you need to do is to implement the following macros :
	- IS_IN_PORTX_RANGE
	- GET_PORTX_BASE

Signed-off-by: Paul Semel <semelpaul@gmail.com>

### CODE REVIEW

File: src/digital.h [hunk 1]
```diff
@@ -29,6 +30,39 @@
 
 #define LED 13
 
+#define IS_IN_PORTD_RANGE(pin) pin <= 7
```
Comment (Keenuts) [inline src/digital.h]: you may want to write it like that:
IS_IN_PORTD_RANGE(pin)    (pin <= 7)

same for the ones under

File: src/digital.h [hunk 2]
```diff
@@ -29,6 +30,39 @@
...
+#define ANALOG_8 90
+#define ANALOG_9 91
+#define ANALOG_10 92
+#define ANALOG_11 93
+#define ANALOG_12 94
+#define ANALOG_13 95
+#define ANALOG_14 96
+#define ANALOG_15 97
+
+#define IS_IN_PORTF_RANGE(pin) pin >= ANALOG_0 && pin <= ANALOG_15
```
Comment (Keenuts) [inline src/digital.h]: same here (parenthesis)

File: src/digital.c [hunk 3]
```diff
@@ -1,52 +1,140 @@
 #include "digital.h"
+#include <stdbool.h>
 
-void digital_write(uint8_t pin, uint8_t value)
+static void pinWrite(uint8_t pin, uint8_t value, bool dir)
```
Comment (Keenuts) [inline src/digital.c]: same prototype model maybe (lowercase, underscore)

Comment (Keenuts) [inline src/digital.c]: parameter name ?
The pinWrite sounds weird with the direction param
maybe call is 'set_dir' or something less ambiguous ?

**Review by Keenuts — COMMENTED** [review]
great, just some small fix to do

File: src/digital.h (hunk 1 above)
Comment (paulsemel) [reply src/digital.h]: Acutally, you want to write it like this:
```c
IS_IN_PORTX_RANGE(pin) ((pin) <= 7)
```
But you have a point here.

File: src/digital.c (hunk 3 above)
Comment (paulsemel) [reply src/digital.c]: Sure

**Review by paulsemel — COMMENTED** [review]
Ack

**Review by paulsemel — COMMENTED** [review]
Applied changes

**Review by Keenuts — APPROVED** [review]

**Pull request merged**